src/main/java/picocli/CommandLine.java

```diff
@@ -4799,6 +4822,28 @@ private static String format(String formatString, Object... params) {
         public String[] getVersion() throws Exception { throw new UnsupportedOperationException(); }
     }
 
+    /**
+     * Provides a way to modify how the command model is built.
+     * This is useful for applications that need to modify the model dynamically depending on the runtime environment.
+     * <p>
+     * Commands may configure a model transformer using the
+     * {@link Command#modelTransformer()} annotation attribute, or via the
+     * {@link CommandSpec#modelTransformer(IModelTransformer)} programmatic API.
+     * <p>
+     * The transformers are invoked only once, after the full command hierarchy is constructed.
+     */
+    public interface IModelTransformer {
```

Comment:
Let's add a `@since 4.6` tag to this javadoc.

Reply:
Ok.